src/openrct2/object/RideObject.cpp

```diff
@@ -83,8 +83,8 @@ void RideObject::ReadLegacy(IReadObjectContext* context, IStream* stream)
     _legacyType.max_height = stream->ReadValue<uint8_t>();
     // Skipping a uint64_t for the enabled track pieces and two uint8_ts for the categories.
     stream->Seek(10, STREAM_SEEK_CURRENT);
-    _legacyType.shop_item[0] = stream->ReadValue<uint8_t>();
-    _legacyType.shop_item[1] = stream->ReadValue<uint8_t>();
+    _legacyType.shop_item[0] = stream->ReadValue<ShopItem>();
```

Comment:
Change this back to `uint8_t`. We might change the shop item to use uint16_t at some point, but in .DAT objects it will always be a `uint8_t`.

Reply:
Do you mean `_legacyType.shop_item` to be `uint8_t` or only `ReadValue<>`?
I would like to do following: `_legacyType.shop_item[0] = static_cast<ShopItem>(stream->ReadValue<uint8_t>());`

Reply:
Only `ReadValue`.

The code you suggested look good to me.